### PR TITLE
Add closemark to navigator view

### PR DESCRIPTION
**Pull request by SamiPuttonen**
## Proposed Changes

  - Closemark and functionality to Navigator view. 
  - image

## Pull Request Check List

  - A reasonable set of unit tests is included
  - Console does not show new warnings/errors
  - Changes are documented or they are self explanatory
  - This pull request does not have any merge conflicts
  - All existing tests pass in CI build

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform
  - Check that the implementation matches the design, when such one is defined in a Jira issue
  - Merge the pull request

### CODE REVIEW

File: app/component/itinerary/Navigator.js [hunk 1]
```diff
@@ -33,6 +34,19 @@ function Navigator({
 
   return (
     <div>
+      <div className="navigator-top-section">
+        <button
+          type="button"
+          onClick={() => {
+            if (window && window.location) {
```
Comment (vesameskanen) [inline app/component/itinerary/Navigator.js]: Do no reload page.  It is slow and not necessary. 

Navigator visibility is controlled by itinerary page state and can be changed using the function prop.

File: app/component/itinerary/Navigator.js [hunk 2]
```diff
@@ -33,6 +34,19 @@ function Navigator({
...
+              window.location.reload();
+            }
+          }}
+          className="close-button cursor-pointer"
+        >
+          <Icon img="icon-icon_close" className="medium" />
```
Comment (vesameskanen) [inline app/component/itinerary/Navigator.js]: Close cross is not black, it probably obeys some theme color (primary_color ?)

File: app/component/itinerary/navigator.scss [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.navigator-top-section {
+  margin: 0.25em;
+  position: absolute;
```
Comment (vesameskanen) [inline app/component/itinerary/navigator.scss:2]: Why mixture of em and pixel values?

**Review by vesameskanen — CHANGES_REQUESTED** [review]

File: app/component/itinerary/Navigator.js (hunk 2 above)
Comment (SamiPuttonen) [reply app/component/itinerary/Navigator.js]: Clearing style medium off and and new color attribute to navigator-top-section -style.

File: app/component/itinerary/navigator.scss (hunk 3 above)
Comment (SamiPuttonen) [reply app/component/itinerary/navigator.scss:2]: That style class was removed.

File: app/component/itinerary/Navigator.js (hunk 1 above)
Comment (SamiPuttonen) [reply app/component/itinerary/Navigator.js:41]: Give me here example how we do this, please? I have a try to do this same way than we did at NavigatorButton, but it didn't work.

Comment (SamiPuttonen) [reply app/component/itinerary/Navigator.js]: Can you check this functionality? I made little change to it.

File: app/component/itinerary/ItineraryPage.js [hunk 4]
```diff
@@ -1227,9 +1230,11 @@ ItineraryPage.propTypes = {
   relayEnvironment: relayShape.isRequired,
   mapLayers: mapLayerShape.isRequired,
   mapLayerOptions: mapLayerOptionsShape.isRequired,
+  setNavigation: PropTypes.func,
```
Comment (vesameskanen) [inline app/component/itinerary/ItineraryPage.js]: Why did you add this? Please read some tutorials about react components and their props.

File: app/component/itinerary/ItineraryPage.js [hunk 5]
```diff
@@ -1067,6 +1067,9 @@ export default function ItineraryPage(props, context) {
           focusToPoint={focusToPoint}
           focusToLeg={focusToLeg}
           relayEnvironment={props.relayEnvironment}
+          buttonClickAction={() => {
```
Comment (vesameskanen) [inline app/component/itinerary/ItineraryPage.js]: Look how Navigator view is opened. Close it exactly the same way.

File: app/component/itinerary/navigator.scss [hunk 6]
```diff
@@ -0,0 +1,6 @@
+.navigator-top-section {
+  position: absolute;
+  right: 20px;
+  top: 12px;
+  color: black;
+}
```
Comment (vesameskanen) [inline app/component/itinerary/navigator.scss:9]: Sorry for a unclear review note earlier - I meant that cross should not be black. It obeys a theme color, in HSL case a dark blue tone. Study how theme colors are used. We rarely use hard coded color values.

File: app/component/itinerary/Navigator.js [hunk 7]
```diff
@@ -21,26 +19,37 @@ function Navigator({
...
   }, [time]);
 
   return (
     <div>
-      Tracking {itinerary.legs.length} legs, current {currentLeg?.mode}
+      <div className="navigator-top-section">
+        <button
+          type="button"
+          onClick={e => props.buttonClickAction(e)}
+          className="close-button cursor-pointer"
```
Comment (vesameskanen) [inline app/component/itinerary/Navigator.js:43]: We need aria label here.

**Review by vesameskanen — CHANGES_REQUESTED** [review]

File: app/component/itinerary/navigator.scss (hunk 6 above)
Comment (SamiPuttonen) [reply app/component/itinerary/navigator.scss:9]: Yes, I will do. But first I remove that color off there.

File: app/component/itinerary/Navigator.js (hunk 7 above)
Comment (SamiPuttonen) [reply app/component/itinerary/Navigator.js:43]: I knew it, but some example we didn't have it.. But I add that one.

File: app/component/itinerary/ItineraryPage.js (hunk 4 above)
Comment (SamiPuttonen) [reply app/component/itinerary/ItineraryPage.js]: I was testing something what I don't remember anymore. I clean up.

File: app/translations.js [hunk 8]
```diff
@@ -2497,6 +2498,7 @@ const translations = {
     navigate: 'Navigoi',
     'navigation-description': 'Löydä perille ohjatusti!',
     'navigation-header': 'Matkan seuranta',
+    'navigation-label-close': 'Sulje navigaattori',
     nearest: 'Lähimmät {mode}',
```
Comment (vesameskanen) [inline app/translations.js:2502]: Navigaattori on ohjelmointitermi. Parempi kuvaus on esim 'Lopeta matkaopastus'. Näihin saadaan varmaan tarkennuksia mutta pyritään silti tekemään parhaamme jo tässä vaiheessa.

File: app/component/itinerary/Navigator.js [hunk 9]
```diff
@@ -1,11 +1,12 @@
 import PropTypes from 'prop-types';
 import React, { useEffect, useState } from 'react';
+import { intlShape } from 'react-intl';
 import { createFragmentContainer, graphql } from 'react-relay';
 import { itineraryShape } from '../../util/shapes';
 import { legTime } from '../../util/legUtils';
 import Icon from '../Icon';
 
-function Navigator(props) {
+function Navigator(props, context) {
```
Comment (vesameskanen) [inline app/component/itinerary/Navigator.js]: Palauta proppien destrukturointi. Sitä käytetään tarkoituksellisesti koodin luettavuuden parantamiseksi funktiokomponenteissa.

File: app/component/itinerary/navigator.scss (hunk 6 above)
Comment (vesameskanen) [reply app/component/itinerary/navigator.scss:9]: Sulkemisruksi on edelleen väärän värinen.

File: app/component/itinerary/ItineraryPage.js (hunk 5 above)
Comment (vesameskanen) [reply app/component/itinerary/ItineraryPage.js]: setNavigation={setNavigation}

**Review by vesameskanen — CHANGES_REQUESTED** [review]
